src/core.js

```diff
@@ -94,8 +94,14 @@ class Core {
    */
   run(runFor) {
     const stopAt = this._now + runFor;
-    while (this._now < stopAt && this.queue.size() > 0) {
+    while (this.queue.size() > 0) {
       const [when, what] = this.queue.deq();
+      if (when > stopAt) {
```

Comment:
What is `when` and `what`?

Reply:
I didn't comment this file very well, but `this.queue` is a priority queue of future events (functions to call, `what`), sorted by timestamp (`when`).  So this pops the next item off `this.queue` and looks at the time the event is supposed to happen.  If that's after this run is supposed to stop, then we put it back on the queue and bail out.

I changed this bit in this PR because otherwise it would run an event with `when === Infinity`!